main.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// File represents infromation about file
+// we are going to download
+type File struct {
+	io.Reader // To implement Reader interface
+	name      string
+	totalSize int64 // Total content size in bytes
+	counter   int64 // Total # of bytes transferred
+}
+
+// parseURL takes string value link that must be a correct URL,
+// parses it and returns the same link with nil error if all is ok
+// and empty string with corresponding error if something is wrong
+func parseURL(link string) (string, error) {
+	_, err := url.ParseRequestURI(link)
+	if err != nil {
+		return "", err
+	}
+	resp, err := http.Get(link)
+	if err != nil {
+		return "", fmt.Errorf("wget: error connecting url :%v", err)
+	}
+	defer resp.Body.Close()
+
+	bodySize, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 0)
+	if err != nil {
+		return "", fmt.Errorf("wget: content length = %v :%v", bodySize, err)
+	}
+	return link, nil
+}
+
+func main() {
+
+	var linksList []string
+	// parsing urls
+	// if correct, adding them to slice
+	// else print error
+	for _, inputLink := range os.Args[1:] {
+		parsedLink, err := parseURL(inputLink)
+		if err == nil {
+			linksList = append(linksList, parsedLink)
+		} else {
+			fmt.Fprintf(os.Stderr, "wget: error parsing url %v: %v\n", inputLink, err)
+		}
+	}
+
+	var fileList []*File
```

Comment:
at this moment you already know number of links, so why not fileList := make([]*File, Len(linksList)) ?